hw/ip_templates/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -0,0 +1,476 @@
+# Boot, ROM execution and Patching
+
+## Glossary
+
+| Term  | Description                         |
+|-------|-------------------------------------|
+| SoC   | System On Chip                      |
+| DFx   | Design for Test and Debug           |
+| DFT   | Design for Test                     |
+
+## Scope
+
+This document covers the boot mechanism for Integrated OpenTitan.
+This includes the execution of the OpenTitan ROM, associated patch support in efuses / OTP for survivability and the code fetch from external flash.
+
+## Overview
+
+The integrated Open Titan is a root of trust for measurement, integrated within the SoC.
+It provides the framework necessary to bootstrap the SoC computing elements in a secure and trusted fashion.
+This document covers the changes necessary for the OpenTitan to boot first while also allowing an SoC with integrated OpenTitan to add small customizations to the boot path configuration that fit with the SoC topology.
+A few examples are voltage ramp up sequence, clock settings etc.
+Such customizations are expected to
+
+- Be non-proprietary such that they can be made available in the open source environment for audit.
+- Only require minimal flow to enable a path to external flash that is under OpenTitan control.
+
+Flows involving reset and bring up for the rest of the SoC, which may be potentially more complex and involve proprietary code, are expected to be handled by other computing element(s) that are bootstrapped by the integrated OpenTitan and running authenticated / measured code.
+
+This proposal introduces a second OpenTitan boot ROM partition with efuse/OTP based patching mechanism that enables each SoC to add such small customizations and provide critical survivability hooks while also removing the dependency on the integrated flash component.
+Integrated flash technology may not be available for SoCs built using advanced process nodes.
+
+Two models for OpenTitan based ROM patch deployment are envisioned:
+
+- **Factory patching** - here the ROM patches are programmed into the eFuses in the factory.
+  This may be to accommodate post-silicon bug fixes without needing a new step of the silicon.
+  Performed through the secure manufacturing flow.
+- **In-field patching** - The capability may be used to recover parts in the field but that requires an ability to burn eFuses in the field.
+
+Note that patch structure for both deployment models is the same.
+An infield patch can be viewed as a new revision of the OTP patch.
+The patch in eFuses will be signed using an appropriate PK signing algorithm
+
+### Programming method:
+
+- If prior to the PROD lifecycle stage: Using the manufacturing efuse programming interface.
+- In PROD lifecycle stage: Secure, verified patch update programming managed by secure application running on OpenTitan Ibex core (e.g., update patch revision in factory after provisioning, backup in-field patching mechanism).
```

Comment:
Could you try to improve the structure of this text?
- Neither bullet point contains a sentence (and I think they should have at least one each!)
- There are lots of impressive words (secure, verified, secure again) associated with the second point. If there is an example, it needs describing. If not, I suggest dropping some of the adjectives.
- Please add a full stop or two. This is an currently a very long run-on-not-quite-sentence.